### PR TITLE
[WIP] Boost patch

### PR DESCRIPTION
**Pull request by ESeNonFossiIo**
I wrote a test but it should fail. 
Travis has not boost libraries installed.
@asartori86 : can you write me the flag I have to use?

### CODE REVIEW

**Comment by asartori86** [comment]
you should rename the file `directory_00.output` as

```
directory_00.with_boost=on.output
```

@luca-heltai, is it right?

**Comment by luca-heltai** [comment]
Yup. 

Luca

**Comment by luca-heltai** [comment]
Actually, no.

That mechanism works with the libraries for deal.ii not with our own definitions... Even if deal.ii found the boost libraries, they might not be complete, thus we cannot make sure they are ok for us.

Instead of changing directly the cmake cxx flags, can you do as suggested in the comment above? Don't use directly -I and -l flags...

L

**Comment by luca-heltai** [comment]
INCLUDE_DIRECTORIES (your_headers ${BOOST_INCLUDE_DIRS})

And where the other Target link libraries command is:

TARGET_LINK_LIBRARIES (your_lib ${BOOST_LIBRARIES})

**Comment by luca-heltai** [comment]
closes #96

File: CMakeLists.txt [hunk 1]
```diff
@@ -34,11 +34,17 @@ SET(_d2_build_types "Release;Debug")
...
-IF(BOOST_FOUND)
+FIND_PACKAGE(Boost COMPONENTS system filesystem)
+IF(Boost_FOUND)
+  # add_definitions(${Boost_LIB_DIAGNOSTIC_DEFINITIONS})
+  SET(Boost_USE_STATIC_LIBS OFF)
+  SET(Boost_USE_MULTITHREADED ON)
+  SET(Boost_USE_STATIC_RUNTIME OFF)
   ADD_DEFINITIONS(-DDEAL_II_SAK_WITH_BOOST)
+  # INCLUDE_DIRECTORIES (${Boost_LIBRARY_DIR})
+  # TARGET_LINK_LIBRARIES (${TARGET_LIB} ${Boost_LIBRARIES} )
```
Comment (ESeNonFossiIo) [inline CMakeLists.txt:45]: There are problems here. Using commented lines I get the following error:

``` bash
  Cannot specify link libraries for target "dealii-sak" which is not built by
  this project.
```

**Comment by luca-heltai** [comment]
In fact you have to do this twice, once for the debug library and once for the release library. Look for the line ADD_LIBRARY in the file, and add it there, copying what is already there, i.e., under

```
ADD_LIBRARY(${_lib} SHARED ${_library_source_files})
if(Boost_FOUND)
   TARGET_LINK_LIBRARIES(${_lib} ${Boost_LIBRARIES})
endif()
```

**Comment by luca-heltai** [comment]
Also make sure that the tester works! If you cannot make one unique test, don't make a test for the configuration phase.

**Comment by ESeNonFossiIo** [comment]
Probably I did not understood what do you mean but in this way there are plenty errors.
It does not find libraries or it complains for boost_filesystem and boost_system or the number of variables it is not correct...

**Pull request closed without merging**